sqlglot/dialects/snowflake.py

```diff
@@ -190,6 +190,13 @@ def _parse_regexp_replace(args: t.List) -> exp.RegexpReplace:
     return regexp_replace
 
 
+def _show_parser(*args: t.Any, **kwargs: t.Any) -> t.Callable[[Snowflake.Parser], exp.Show]:
+    def _parse(self: Snowflake.Parser) -> exp.Show:
+        return self._parse_show_snowflake(*args, **kwargs)
+
+    return _parse
+
+
```

Comment:
This is basically what we do for MySQL; since it's just these two dialects atm, I thought I'd avoid introducing more abstractions to dry this out. Perhaps we can move these duplicated chunks of logic in the base parser if / when we want to add `SHOW` parsing for a third dialect?